tools/m2kcli/commands/analog/analog_out.cpp

```diff
@@ -0,0 +1,278 @@
+				addOutputMessage(output,
+						 ("sampling_frequency_channel_" + std::to_string(channel)).c_str(),
+						 std::to_string(analogOut->getSampleRate(channel)));
+				addOutputMessage(output,
+						 ("oversampling_ratio_channel_" + std::to_string(channel)).c_str(),
+						 std::to_string(analogOut->getOversamplingRatio(channel)));
+			}
+		} else {
+			throw std::runtime_error("Invalid attribute: " + argument + '\n');
+		}
+		index++;
+	}
+}
+
+void AnalogOut::handleSetChannel()
+{
+	std::vector<int> channels;
+	Validator::validate(std::string(optarg), "channel", channels);
+	int index = optind;
+	while (index < argc && *argv[index] != '-') {
+		std::string argument(argv[index]);
+		if (argument.find('=') == std::string::npos) {
+			throw std::runtime_error("Expecting the following format: <attribute>=<value>\n");
+		}
+		std::istringstream iss(argument);
+		std::string value;
+		getline(iss, argument, '=');
+		getline(iss, value, '=');
+
+		if (argument == "sampling_frequency") {
+			for (int &channel : channels) {
+				analogOut->setSampleRate(channel, std::stod(value));
+			}
+		} else if (argument == "oversampling_ratio") {
+			for (int &channel : channels) {
+				analogOut->setOversamplingRatio(channel, std::stod(value));
+			}
+		} else {
+			throw std::runtime_error("Invalid attribute: " + argument + '\n');
+		}
+		index++;
+	}
+}
+
+const struct option AnalogOut::options[] = {
+	{"help",        no_argument,       nullptr, 'h'},
+	{"quiet",       no_argument,       nullptr, 'q'},
+	{"calibrate",   no_argument,       nullptr, 'C'},
+	{"generate",    required_argument, nullptr, '9'},
+	{"get-channel", required_argument, nullptr, 'G'},
+	{"set-channel", required_argument, nullptr, 'S'},
+	{nullptr, 0,                       nullptr, 0},
+};
+
+
+const char *const AnalogOut::helpMessage = "Usage:\n"
+					   "m2kcli analog-out [uri]\n"
+					   "                  [-h | --help]\n"
+					   "                  [-q | --quiet]\n"
+					   "                  [-C | --calibrate]\n"
+					   "                  [-9 | --generate channel=<index>,... cyclic=<value> raw=<value> [format=<type>] [file=<path>]]\n"
+					   "                  [-G | --get-channel channel=<index> <attribute> ...]\n"
+					   "                  [-S | --set-channel channel=<index> <attribute>=<value> ...]\n"
+					   "\n"
+					   "Pinout:\n"
+					   "\u25A1 \u25A1 \u25A1 \u25A1 \u25A4 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					   "\u25A1 \u25A1 \u25A1 \u25A1 \u25A5 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1 \u25A1\n"
+					   "Positional arguments:\n"
+					   "  uri                   describe the context location \n"
+					   "                        auto | ip:192.168.2.1 | usb:XX.XX.X\n"
+					   "Optional arguments:\n"
+					   "  -h, --help            show this help message and exit\n"
+					   "  -q, --quiet           return result only\n"
+					   "  -C, --calibrate       calibrate the DAC"
```

Comment:
Nothing functional. Just a missing "\n".

Reply:
The help message was corrected in commit a61547e.